napalm_eos/eos.py

```diff
@@ -1090,35 +1093,43 @@ def get_route_to(self, destination='', protocol=''):
         return routes
 
     def get_snmp_information(self):
+        """get_snmp_information() for EOS.  Re-written to not use TextFSM"""
 
-        snmp_information = dict()
-
-        commands = list()
-        commands.append('show running-config | section snmp-server')
-        raw_snmp_config = self.device.run_commands(commands, encoding='text')[0].get('output', '')
-
-        snmp_config = napalm_base.helpers.textfsm_extractor(self, 'snmp_config', raw_snmp_config)
-
-        if not snmp_config:
-            return snmp_information
-
-        snmp_information = {
-            'contact': py23_compat.text_type(snmp_config[0].get('contact', '')),
-            'location': py23_compat.text_type(snmp_config[0].get('location', '')),
-            'chassis_id': py23_compat.text_type(snmp_config[0].get('chassis_id', '')),
+        # Default values
+        snmp_dict = {
+            'chassis_id': '',
+            'location': '',
+            'contact': '',
             'community': {}
         }
 
-        for snmp_entry in snmp_config:
-            community_name = py23_compat.text_type(snmp_entry.get('community', ''))
-            if not community_name:
-                continue
-            snmp_information['community'][community_name] = {
-                'acl': py23_compat.text_type(snmp_entry.get('acl', '')),
-                'mode': py23_compat.text_type(snmp_entry.get('mode', 'ro').lower())
-            }
+        commands = [
+            'show snmp chassis',
+            'show snmp location',
+            'show snmp contact'
+        ]
+        snmp_config = self.device.run_commands(commands, encoding='json')
+        for line in snmp_config:
+            for k, v in line.items():
+                if k == 'chassisId':
+                    snmp_dict['chassis_id'] = v
+                else:
+                    snmp_dict[k] = v
```

Comment:
Yes, I agree.

Just please strip the `"` at the beginning and at the end  - if double qoutes happen to not be there, won't hurt.